litestar/serialization.py

```diff
@@ -125,6 +129,38 @@ def default_serializer(value: Any, type_encoders: Mapping[Any, Callable[[Any], A
     raise TypeError(f"Unsupported type: {type(value)!r}")
 
 
+PydanticUUIDType = TypeVar("PydanticUUIDType", bound="UUID1")
+
+
+def _dec_pydantic_uuid(type_: type[PydanticUUIDType], val: Any) -> PydanticUUIDType:
```

Comment:
This logic is copied out of pydantic's validators for UUID types - I'd imagine if we can see this to the end, then hooks like this could be added by plugin if users were working with pydantic.